MobileSyncExplorerKotlinTemplate/app/src/main/java/com/salesforce/mobilesyncexplorerkotlintemplate/core/extensions/CoroutineExt.kt

```diff
@@ -0,0 +1,7 @@
+package com.salesforce.mobilesyncexplorerkotlintemplate.core.extensions
+
+import kotlinx.coroutines.sync.Mutex
+
+fun Mutex.requireIsLocked() {
+    require(isLocked) { "This operation is only permitted while this mutex is locked." }
```

Comment:
I like the idea of crashing the app when developing since it gives immediate feedback to the developer developer that something needs fixing, but I agree that crashing in every case may not be the best choice.  Maybe only crash if `BuildConfig.DEBUG` is enabled, otherwise log an error?

Reply:
It definitely makes sense during development.  I would much prefer the template log than throw, even for debug.